pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -444,6 +455,10 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 			} else {
 				con.mu.Unlock()
 			}
+		case updateEv := <-con.updateChannel:
+			if updateEv.workloadLabel && con.modelNode != nil {
+				con.modelNode.WorkloadLabels = nil
```

Comment:
setting to nil so it gets updated later seems kind of hacky, can we properly update it here?

Reply:
make sense